src/tests/Regressions/coreclr/GitHub_42732/test42732.cs

```diff
@@ -0,0 +1,40 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+using System;
+using System.Runtime.CompilerServices;
```

Comment:
I don't think this test is appropriate. The github number thing here makes sense with the JIT tests, but not with defined behaviors. I'd prefer to see this test added to an existing collection for `TypedReference` under `src/tests/Loader/classloader` or, since this is pure managed, in [`TypedReferenceTests.cs`](https://github.com/dotnet/runtime/blob/main/src/libraries/System.Runtime/tests/System/TypedReferenceTests.cs).

Reply:
Reasonable. I originally put the test here, as I guessed it was weird optimizer thing.... as it was only reproducing on Linux then I looked a bit deeper.

Reply:
Instead of moving this, I'd also be fine with clearly documenting in the test the state we are validating here.